core/svelte-compiler.js

```diff
@@ -74,8 +75,8 @@ SvelteCompiler = class extends CachingCompiler {
     try {
       const compiled = svelte.compile(raw, {
         filename: path,
-        name: file.getBasename()
-          .slice(0, -(extension.length + 1)) // Remove extension
+        name: basename
+          .slice(0, basename.indexOf('.')) // Remove extension
```

Comment:
Shouldn't this be `lastIndexOf` to be sure we just get the extension and not everything after the first `.`?
`foo.bar.html` becomes `foo` instead of `foo.bar` as the code is now, right?